tokio-util/src/sync/cancellation_token.rs

```diff
@@ -66,6 +67,19 @@ pin_project! {
     }
 }
 
+/// A Future that is resolved once the corresponding [`CancellationToken`]
+/// is cancelled.
+///
+/// This is counterpart to [`WaitForCancellationFuture`] where it takes
+/// [`CancellationToken`] by value instead of using a reference.
+#[must_use = "futures do nothing unless polled"]
+pub struct WaitForCancellationFutureOwned {
+    cancellation_token: CancellationToken,
+
+    /// Use 'static lifetime here because we don't have 'this.
+    future: Option<mem::ManuallyDrop<tokio::sync::futures::Notified<'static>>>,
+}
```

Comment:
Honestly, we can just do this. The `ManuallyDrop` isn't necessary.
```suggestion
#[must_use = "futures do nothing unless polled"]
pub struct WaitForCancellationFutureOwned {
    // Since `future` is the first field, it is dropped before the cancellation token.
    future: Option<tokio::sync::futures::Notified<'static>>,
    cancellation_token: CancellationToken,
}
```

Reply:
In fact, we could even drop the `Option` if we want. The `Notify` is inside a heap allocation, so it's `&Notify` isn't dangling even if we move the future around before it is pinned.

Reply:
> In fact, we could even drop the `Option` if we want. The `Notify` is inside a heap allocation, so it's `&Notify` isn't dangling even if we move the future around before it is pinned.

I've dropped the use of `Option` as it makes `WaitForCancellationFutureOwned` more similar to `WaitForCancellationFuture` and easier to maintain.